### PR TITLE
DG-19 Add JWT token validation

### PR DESCRIPTION
**Pull request by Mickss**

### CODE REVIEW

File: src/main/java/org/axion/axion_api_gateway/RequestValidator.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    private boolean validateTokenWithAuthService(String token) {
+        OkHttpClient client = new OkHttpClient();
+
+
+        String json = createJson(token);
+        RequestBody body = RequestBody.create(json, JSON);
+
+        Request request = new Request.Builder()
+                .url("http://localhost:25003/auth/validate")
```
Comment (tknowak) [inline src/main/java/org/axion/axion_api_gateway/RequestValidator.java]: This port (25003) should be taken from appConfig, not hardcoded.

File: src/main/java/org/axion/axion_api_gateway/RequestValidator.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    public String createJson(String token) {
+        return "{"
+                + "\"token\":\"" + token + "\""
+                + "}";
+    }
+
+    private boolean validateTokenWithAuthService(String token) {
+        OkHttpClient client = new OkHttpClient();
+
+
```
Comment (tknowak) [inline src/main/java/org/axion/axion_api_gateway/RequestValidator.java]: No need for double empty lines 🙂

**Review by tknowak — CHANGES_REQUESTED** [review]

File: src/main/java/org/axion/axion_api_gateway/RequestValidator.java (hunk 1 above)
Comment (Mickss) [reply src/main/java/org/axion/axion_api_gateway/RequestValidator.java]: Fixed

File: src/main/java/org/axion/axion_api_gateway/RequestValidator.java (hunk 2 above)
Comment (Mickss) [reply src/main/java/org/axion/axion_api_gateway/RequestValidator.java]: Fixed

File: src/main/java/org/axion/axion_api_gateway/RequestValidator.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        String token = header.substring("Bearer ".length());
+
+        boolean isTokenValid = validateTokenWithAuthService(token);
+        if (!isTokenValid) {
+            throw new RequestValidationException("Invalid token", HttpStatus.UNAUTHENTICATED);
+        }
+    }
+
+    public String createJson(String token) {
```
Comment (tknowak) [inline src/main/java/org/axion/axion_api_gateway/RequestValidator.java]: This method should be private, it's only for this class context.

**Review by tknowak — CHANGES_REQUESTED** [review]

File: src/main/java/org/axion/axion_api_gateway/RequestValidator.java (hunk 3 above)
Comment (Mickss) [reply src/main/java/org/axion/axion_api_gateway/RequestValidator.java]: Fixed

**Review by tknowak — APPROVED** [review]

**Pull request merged**